table.go

```diff
@@ -57,6 +60,48 @@ func NewRoutingTable(bucketsize int, localID ID, latency time.Duration, m peerst
 	return rt
 }
 
+func (rt *RoutingTable) GetAllBuckets() []*Bucket {
+	rt.tabLock.RLock()
+	defer rt.tabLock.RUnlock()
+	return rt.Buckets
```

Comment:
Can we add a comment explaining why this is safe (it's append-only).

Reply:
Done